### ISSUE TITLE
handleUpload($event) gets triggerd twice

### ISSUE THREAD

**Issue by maartentibau** [jkuri/ngx-uploader#207]
Hi,

I'm using the more "advanced" way of your ngx-uploader. So I have the `autoUpload` set to `false`.
And when I click a button, I trigger the `startUpload` event.  The thing is, the `handleUpload` method is triggered twice.. I have no idea why this is happening to be honest.

The code I'm using is almost exact the same as your example.

**Comment by lagfvu** [comment]
@maartentibau please did you find a solution, i'm having the same issue

**Comment by maartentibau** [comment]
Nope... I have no idea what is going wrong. I also haven't found the time to dive into the source code sadly 😞 I was hoping @jkuri might have an idea what it could be. I updates to the latest version, but that didn't solve it.

**Comment by andershagebakken** [comment]
I have a similar issue with the advance example, but it gets fired four times.

**Comment by maartentibau** [comment]
OK, I figured out what's actually causing these multiple uploads. I haven't looked into the source code, but to trigger multiple uploaded.. just try this.

1. select a file
2. select another file
3. select yet another file
4. Yes.. you guessed it.. select another file

Then click on your upload button, you'll notice multiple uploads being fired. So... for some reason the "select a file" events are being stored and on upload all fired again, pretty strange behavior.

Gonna look into the source code after this weekend trying to figure out how to solve this. Keep you guys posted.

**Comment by yonimoses** [comment]
@maartentibau  - any luck with the source code? I'm also getting multiple times fired . it wasn't like that in v1+ , it started when I switched to version 2.

**Comment by jculverwell** [comment]
I am also getting this.  Anyone got any ideas of how to work around this?

**Comment by jculverwell** [comment]
Ok, so found why is was getting called twice for me, on the setTimeout I had added 3 seconds as I wanted to show a message for a few seconds.  Removing the three seconds (3000 below) fixed the issue.

` handleUpload(data): void {
    setTimeout(() => {
      this.zone.run(() => {
        if (data && data.response) {
              alert('handle upload');
             console.log(data);
       }
      });
    }, 3000);`
}
